packages/terra-form-input/src/FormInput.jsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './FormInput.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The defaultValue of the input field. Use this to create an uncontrolled input.
+   */
+  defaultValue: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+  /**
+   * Whether the input is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Whether the input is invalid
+   */
+  isInvalid: PropTypes.bool,
+  /**
+   * Function to trigger when the input loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Function to trigger when user changes the input value. Provide a function to create a controlled input.
+   */
+  onChange: PropTypes.func,
+  /**
+   * Function to trigger when the input gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Content to be displayed as the name.
+   */
+  name: PropTypes.string,
+  /**
+   * Whether the input is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the input field. Use this to create a controlled input.
+   */
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+};
+
+const defaultProps = {
+  defaultValue: undefined,
+  disabled: false,
+  isInvalid: false,
+  onBlur: undefined,
+  onChange: undefined,
+  onFocus: undefined,
+  name: null,
+  required: false,
+  value: undefined,
+};
+
+class FormInput extends React.Component {
+  render() {
+    const {
+      defaultValue,
+      disabled,
+      isInvalid,
+      onBlur,
+      onChange,
+      onFocus,
+      name,
+      required,
+      value,
+      ...customProps
+    } = this.props;
+
+    const attributes = Object.assign({}, customProps);
+    const formInputClassNames = cx([
+      'form-input',
+      { 'form-error': isInvalid },
+      attributes.className,
+    ]);
+
+    if (required) {
+      attributes['aria-required'] = 'true';
+    }
+
+    if (value !== undefined) {
+      attributes.value = value;
+    } else {
+      attributes.defaultValue = defaultValue;
+    }
+
+    return (
+      <input
+        aria-label="input label"
+        name={name}
+        onBlur={onBlur}
+        onChange={onChange}
+        onFocus={onFocus}
+        ref={(input) => { this.textInput = input; }}
+        disabled={disabled}
+        required={required}
+        {...attributes}
```

Comment:
We don't appear to be consistent throughout all of the components but I believe the current standard for additional attributes is to spread them first so that explicitly declared prop values always win. 
```
      <input
        {...attributes}
        aria-label="input label"
        name={name}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
        disabled={disabled}
        required={required}
        className={formInputClassNames}
      />
```

Reply:
I agree I had been following the example in `terra-form-textarea`. If this is the standard we want to follow, we should probably log an issue to update it [here](https://github.com/cerner/terra-core/blob/Form-Input-Element/packages/terra-form-textarea/src/Textarea.jsx#L164).

Reply:
Done. cd36335322ce48d3beb9b5e3edce6c8a899ae956